Src/WitsmlExplorer.Desktop/src/main/main.ts

```diff
@@ -69,6 +77,18 @@ function showErrorAndQuit(message: string) {
   app.quit();
 }
 
+function showUnfinishedJobsWarningOnClose() {
+  const options: MessageBoxOptions = {
+    type: "warning",
+    buttons: ["Cancel", "Quit"],
+    title: "WARNING",
+    message: "Unfinished jobs will be terminated",
+    detail:
+      "You have unfinished jobs that may cause issues if not completed before exiting.\n\nAre you sure you want to exit the application?"
+  };
+  return dialog.showMessageBoxSync(null, options);
```

Comment:
I suggest attaching the dialog to the mainWindow, which forces it to be handled before doing anything else with the application. Then I also think we can remove the isUnfinishedJobsWarningDialogOpen variable as it's not possible to try to close the window again?

Reply:
(Just replace null with mainWindow)

Reply:
Good catch, I agree. Fixed.